vllm/model_executor/layers/fused_moe/layer.py

```diff
@@ -812,6 +847,7 @@ def __init__(
         assert quant_method is not None
         assert isinstance(quant_method, FusedMoEMethodBase)
         self.quant_method = quant_method
+        self.quant_method.moe = moe
```

Comment:
This seems a bit sketchy to me.  If the `quant_method` needs a `MoEConfig` it should be part of the constructor or passed around as an argument.